src/transformStateWithClones.js

```diff
@@ -7,7 +7,67 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const properties = {
+    addProperties: 'addProperties',
+    clear: 'clear',
+    removeProperties: 'removeProperties',
+  };
+  const newObj = { ...state };
+  let obj = {};
+  const newArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case properties.addProperties:
+        assignNewDataToState(newObj, action);
+        obj = writeToArray(obj, newObj, newArray);
```

Comment:
to complicated
you can just mutate your initial copy (`newObject`) and then push his copy to array

Reply:
@TarasHoliuk if i do newArray.push(copyOfObject)
then the array will be overwritten. And at the end the array will be with 3 identical objects.
Or I misunderstood

Reply:
I changed newObject to copyOfObject